### ISSUE TITLE
Async method should use async benefits in ClientBaseAsync

### ISSUE THREAD

**Issue by force-net** [hflabs/dadata-csharp#30]
Method is declared as async, but in reality it use only one async operation. Main read is sync.

You should use ReadToEndAsync here 
https://github.com/hflabs/dadata-csharp/blob/47583a907082bc88460182509ae663aab5f2bb9f/Dadata/ClientBaseAsync.cs#L92

Also, recommended way for async deserialization from Newtonsoft.Json team is
`await Task.Run( () => JsonConvert.DeserializeObject<T>(responseText));`

Otherwise, all deserialization will be sync and in async implementation will block all execution without any benefit of async version

**Comment by milovidov983** [comment]
> Method is declared as async, but in reality it use only one async operation. Main read is sync.
> 
> You should use ReadToEndAsync here
> 
> https://github.com/hflabs/dadata-csharp/blob/47583a907082bc88460182509ae663aab5f2bb9f/Dadata/ClientBaseAsync.cs#L92
> 
> Also, recommended way for async deserialization from Newtonsoft.Json team is `await Task.Run( () => JsonConvert.DeserializeObject<T>(responseText));`
> 
> Otherwise, all deserialization will be sync and in async implementation will block all execution without any benefit of async version

Maybe 
```csharp
/// from
string responseText = r.ReadToEnd(); 
/// to
string responseText = await r.ReadToEndAsync(); 
```
?

https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoendasync?view=net-6.0

**Comment by nalgeon** [comment]
Thanks! Fixed in [21.12.0](https://github.com/hflabs/dadata-csharp/releases/tag/21.12.0).
